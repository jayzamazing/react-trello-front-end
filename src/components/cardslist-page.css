.cardslist-form-area {
  color: black;
}
.cardslist {
  height: 93%;
}
.cardslist .row, .cardslist-form-area {
  height: 100%;
}
