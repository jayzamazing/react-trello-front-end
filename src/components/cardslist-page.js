import React from 'react';
import {connect} from 'react-redux';
import {Redirect} from 'react-router-dom';
import CardslistForm from './cardslist-form';
import './cardslist-page.css';

export function CardslistPage(props) {
    // Only visible to logged in users
    if (!props.loggedIn) {
        return <Redirect to="/" />;
    }
    return (
        <div className="cardslist container-fluid">
          <div className="row">
            <div className="cardslist-form-area">
              <CardslistForm match={props.match}/>
            </div>
          </div>
        </div>
    );
}

const mapStateToProps = state => ({
    loggedIn: state.auth.currentUser !== null
});

export default connect(mapStateToProps)(CardslistPage);
