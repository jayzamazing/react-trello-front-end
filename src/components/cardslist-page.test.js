'use strict';
import React from 'react';
import {shallow} from 'enzyme';
import {CardslistPage} from './cardslist-page';
import {Redirect} from 'react-router-dom';
import toJson from 'enzyme-to-json';

describe('cardslist-page', () => {
  it('should render page when authenticated', () => {
    const wrapper = shallow(<CardslistPage loggedIn={true}/>);
    expect(wrapper.find(Redirect)).toHaveLength(0);
  });
  it('authenticated snapshot', () => {
    const wrapper = shallow(<CardslistPage loggedIn={true}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders redirect when user is not authenticated', () => {
    const wrapper = shallow(<CardslistPage loggedIn={false}/>);
    expect(wrapper.find(Redirect)).toHaveLength(1);
  });
  it('not authenticated snapshot', () => {
    const wrapper = shallow(<CardslistPage loggedIn={false}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
