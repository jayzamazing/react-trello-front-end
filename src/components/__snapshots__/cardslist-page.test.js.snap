// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cardslist-page authenticated snapshot 1`] = `
<div
  className="cardslist container-fluid"
>
  <div
    className="row"
  >
    <div
      className="cardslist-form-area"
    >
      <withRouter(Connect(Cardslist)) />
    </div>
  </div>
</div>
`;

exports[`cardslist-page not authenticated snapshot 1`] = `
<Redirect
  push={false}
  to="/"
/>
`;
